src/tests/dbus-tests/test_20_LVM.py

```diff
@@ -197,6 +217,102 @@ def test_10_linear(self):
+
+        dbus_vgname = self.get_property(vg, '.VolumeGroup', 'Name')
+        dbus_vgname.assertEqual(vgname)
+
+        first_vdev_uuid = self.get_property(devs[0], '.Block', 'IdUUID').value
+
+        # Create raid1 LV on the VG
+        lvname = 'udisks_test_lv'
+        vg_freesize = self.get_property(vg, '.VolumeGroup', 'FreeSize')
+        vdev_size = vg_freesize.value / len(devs)
+        lv_size = int(vdev_size * 0.75)
+        lv_path = vg.CreatePlainVolumeWithLayout(lvname, dbus.UInt64(lv_size),
+                                                 "raid1", devs[0:2],
+                                                 self.no_options,
+                                                 dbus_interface=self.iface_prefix + '.VolumeGroup')
+        self.assertIsNotNone(lv_path)
+
+        lv = self.bus.get_object(self.iface_prefix, lv_path)
+        self.get_property(lv, '.LogicalVolume', 'SyncRatio').assertEqual(1.0, timeout=60, poll_vg=vg)
+
+        _ret, sys_type = self.run_command('lvs -o seg_type --noheadings --nosuffix %s/%s' % (vgname, lvname))
+        self.assertEqual(sys_type, "raid1")
+        self.get_property(lv, '.LogicalVolume', 'Layout').assertEqual("raid1")
+
+        def assertSegs(struct, size, pv):
+            self.assertEqual(struct["type"], "linear")
+            self.assertNotIn("data", struct)
+            self.assertNotIn("metadata", struct)
+            if pv is not None:
+                self.assertEqual(len(struct["segments"]), 1)
+                if size is not None:
+                    self.assertEqual(struct["segments"][0][1], size)
+                self.assertEqual(struct["segments"][0][2], pv.object_path)
+            else:
+                self.assertEqual(len(struct["segments"]), 0)
+
+        def assertRaid1Stripes(structs, size, pv1, pv2):
+            self.assertEqual(len(structs), 2)
+            assertSegs(structs[0], size, pv1)
+            assertSegs(structs[1], size, pv2)
+
+        def assertRaid1Structure(pv1, pv2):
+            struct = self.get_property(lv, '.LogicalVolume', 'Structure').value
+            self.assertEqual(struct["type"], "raid1")
+            self.assertEqual(struct["size"], lv_size)
+            self.assertNotIn("segments", struct)
+            assertRaid1Stripes(struct["data"], lv_size, pv1, pv2)
+            assertRaid1Stripes(struct["metadata"], None, pv1, pv2)
+
+        def waitRaid1Structure(pv1, pv2):
+            for _ in range(5):
+                try:
+                    assertRaid1Structure(pv1, pv2)
+                    return
+                except AssertionError:
+                    pass
+                time.sleep(1)
+            # Once again for the error message
+            assertRaid1Structure(pv1, pv2)
+
+        waitRaid1Structure(devs[0], devs[1])
+
+        # Yank out the first vdev and repair the LV with the third
+        _ret, _output = self.run_command('echo yes >/sys/block/%s/device/delete' % os.path.basename(self.vdevs[0]))
```

Comment:
This unfortunately yanks out a device that never returns back, failing majority of the other test cases. Contrary to `libblockdev` tests, the LIO target config is set up only once at the testrun startup and torn down at the very end, recycling devices for each testcase. It's just a slightly different approach to the tests which may reveal e.g. bad cleanup.

Reply:
Do you have an idea for how to cope with this?

Reply:
Hmm, and how do the tests even run here?  Where do they get the necessary libblockdev version from?  I am confused...

Reply:
The yanked device can be brought back by scanning the SCSI bus.  I have added code to do that after every test.  It's very quick for the targetd loopback hosts.

Reply:
> The yanked device can be brought back by scanning the SCSI bus. I have added code to do that after every test. It's very quick for the targetd loopback hosts.

Oh, interesting. Can you please move it to the `UdisksLVMTest` class to be on the safe side?

Reply:
> Oh, interesting. Can you please move it to the `UdisksLVMTest` class to be on the safe side?

Done!